packages/trimerge-sync/src/merge-heads.test.ts

```diff
@@ -130,13 +168,17 @@ describe('mergeHeads()', () => {
     const baz = { ref: 'baz', baseRef: 'root' };
     const getCommit = makeGetCommitFn([root, foo, bar, baz]);
     const mergeFn = jest.fn(basicMerge);
-    expect(
-      mergeHeads(['foo', 'bar', 'baz'], basicSort, getCommit, mergeFn),
-    ).toEqual('(root:(root:bar+baz)+foo)');
+    const heads = ['foo', 'bar', 'baz'];
+    expect(mergeHeads(heads, basicSort, getCommit, mergeFn)).toEqual(
+      '(root:baz+(root:foo+bar))',
```

Comment:
foo < bar < baz (given the input order above), so the 'lowest' value merge of all three possibilities is foo+bar, this reduces the tree to baz < [foo+bar] (since the new merge is now the highest value ref) resulting in the final: baz+(foo+bar).  And I think this makes sense: start with the oldest foo, fold in bar, fold in baz.

Reply:
The only thing we might want to consider is, since this is effectively 3 heads with one shared ancestor is if the sequence should actually be (root:(root:foo+bar)+baz)